class MyException < Exception {
    init(msg) {
        super.init(msg);
        this.name="MyException";
    }
}


try {
    var a=1;
    var b=2;
    print "hello";
    print "raising Exception (something happened)";
    raise Exception("something happened");
}
except MyException as e {
    var x=2;
    print "in MyException handler";
    print "oops "+e.name+" "+str(e);
}
except Exception as e {
    var x=2;
    print "in Exception handler";
    print "oops "+e.name+" "+str(e);
}
